### PR TITLE
Make "name" column nullable for workflows

### PR DESCRIPTION
**Pull request by PVince81**
The "name" column is now unused and the code is always inserting an
empty string. While this works with most databases, Oracle complains
because an empty string is equivalent to null.

To fix this, the column definition is changed to allow null values now.

Also added some logging in case of database exceptions, because without
this nothing would be logged to detect the above problem.

### CODE REVIEW

**Comment by PVince81** [comment]
note, I've seen in the JS code that this column is actually legacy and not used any more, see https://github.com/nextcloud/server/blob/master/apps/workflowengine/src/store.js#L105

so probably in the next major version we could remove it completely.

but here I needed a small less impactful fix that we can backport

File: apps/workflowengine/composer/composer/ClassLoader.php [hunk 1]
```diff
@@ -37,13 +37,11 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Jordi Boggiano <j.boggiano@seld.be>
- * @see    https://www.php-fig.org/psr/psr-0/
```
Comment (juliushaertl) [inline apps/workflowengine/composer/composer/ClassLoader.php:40]: Might be that those changes are coming from an outdated composer version that was used to dump the autoloader

File: apps/workflowengine/lib/Migration/Version2200Date20210805101925.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options) {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if ($schema->hasTable('flow_operations')) {
+			$table = $schema->getTable('flow_operations');
+			$table->changeColumn('name', [
+				'notnull' => false,
```
[the remark this answers is not in the document]
Comment (PVince81) [reply apps/workflowengine/lib/Migration/Version2200Date20210805101925.php:48]: will do.

I believe that the schema is going to do a diff after `changeColumn`, so no need to check if the column is nullable already

File: apps/workflowengine/composer/composer/ClassLoader.php (hunk 1 above)
Comment (PVince81) [reply apps/workflowengine/composer/composer/ClassLoader.php]: fixed

File: apps/workflowengine/lib/Migration/Version2200Date20210805101925.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (PVince81) [reply apps/workflowengine/lib/Migration/Version2200Date20210805101925.php:48]: adjusted

**Comment by PVince81** [comment]
```
PHP Warning:  require(/drone/src/lib/composer/composer/platform_check.php): failed to open stream: No such file or directory in /drone/src/lib/composer/composer/autoload_real.php on line 25
```

okay....

**Comment by PVince81** [comment]
weird, so `build/autoloaderchecker.sh` has generated a new file that did not exist before: lib/composer/composer/platform_check.php

I'll git add it but let me know if this is wrong...

**Comment by PVince81** [comment]
failing test is unrelated, the previous run had a different integration test failing

@juliushaertl can you recheck ?

File: lib/composer/composer/platform_check.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+// platform_check.php @generated by Composer
```
Comment (juliushaertl) [inline lib/composer/composer/platform_check.php:3]: @ChristophWurst Any idea why this file is generated out of a sudden? At least my installation does not generate that file even when enforcing it through the composer config.

For reference this seems to be introduced with 2.0 https://php.watch/articles/composer-platform-check

**Comment by PVince81** [comment]
another weird thing is that apps/workflowengine has a composer folder but no composer.json file, so the dump-autoloader function fails with "No lockfile found. Unable to read locked packages".

so how are we supposed to update the autoloader files in apps with `autoloaderchecker.sh` ?

or maybe it's the newer composer that refuses that.
when I run the script it downloads and installs composer 2.1.5.
this is with php 7.4.21

**Comment by PVince81** [comment]
I've asked Arthur to run the script locally and he mentioned no issues when updating, with the same composer version and PHP 7.4.15

so maybe there's something strange in my environment, I'll debug this locally

**Comment by PVince81** [comment]
I've managed to clean the autoloader mess by running the script on a different environment.
For some reason on my local dev env composer behaves differently.

Please recheck

**Review by juliusknorr — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**